src/c/validation_state.cpp

```diff
@@ -7,15 +7,24 @@
 #include "veriblock/assert.hpp"
 #include "veriblock/c/validation_state.h"
 
-const char* VBK_ValidationState_getErrorMessage(VBK_ValidationState* self) {
-  VBK_ASSERT(self);
-  VBK_ASSERT(self->validation_state);
-
-  return self->validation_state->toString().c_str();
+VBK_ValidationState* VBK_NewValidationState() {
+  return new VbkValidationState();
 }
 
-void VBK_ValidationState_Free(VBK_ValidationState* self) {
+void VBK_FreeValidationState(VBK_ValidationState* self) {
   if (self) {
     delete self;
   }
+}
+
+const char* VBK_ValidationState_getErrorMessage(VBK_ValidationState* self) {
+  VBK_ASSERT(self);
+
+  return self->validation_state.toString().c_str();
```

Comment:
this is wrong. toString() returns temporary object and then you return a pointer to tmp object.

store std::string in VBK_ValidationState struct, then string=state.toString() , then return string.c_str()

Reply:
fixed